### ISSUE TITLE
Hub needs ability to search for users outside the currently authenticated user's organization

### ISSUE THREAD

**Issue by rweber-esri** [Esri/arcgis-rest-js#1119]
### Describe the problem

The Hub Discussions feature has the concept of "channels", which is an access control mechanism governing what user's can participate in discussions. Channels currently have an `access` property that can be one of `public|org|private`, and a `groups` property which is an array of ArcGIS Online group IDs that can additionally grant participation privileges to users (groups can be outside the currently auth'd user's organization). User's permitted to participate in a discussion can at-mention other users that are also permitted to participate in a discussion (given the target user's profile visibility is set such that other users, internal, or external to their org can view their record). Given the above, Hub needs the ability to search for users that may be outside the currently authenticated user's organization.

Today, both V3 & V4 of `arcgis-rest-js` only support searching for users from the `/portals/self/users/search` AGO API endpoint, but it seems that endpoint only searches within the currently authenticated user's organization. 

V3:
* [searchUsers](https://github.com/Esri/arcgis-rest-js/blob/97322cabd6ad07bf0b5303b9719d2c085bde3c54/packages/arcgis-rest-portal/src/users/search-users.ts#L36)
* [genericSearch](https://github.com/Esri/arcgis-rest-js/blob/97322cabd6ad07bf0b5303b9719d2c085bde3c54/packages/arcgis-rest-portal/src/util/generic-search.ts#L84)

V4:
* [searchUsers](https://github.com/Esri/arcgis-rest-js/blob/e18091a63755d25acf4cab1c819886b9b06c5d78/packages/arcgis-rest-portal/src/users/search-users.ts#L35)
* [genericSearch](https://github.com/Esri/arcgis-rest-js/blob/e18091a63755d25acf4cab1c819886b9b06c5d78/packages/arcgis-rest-portal/src/util/generic-search.ts#L86)

The [/community/users](https://developers.arcgis.com/rest/users-groups-and-items/user-search.htm) endpoint seems to be able to access user records _outside_ the currently authenticated user's organization.

### Describe the proposed solution

I believe we should either:

1. Add an entirely new search method, e.g. `searchCommunityUsers`, that would pass a new `searchType` of `communityUser` (or similar) to `genericSearch` to route the request to the `/community/users` endpoint.

```ts
// existing method
export function searchUsers(
  search: IUserSearchOptions | SearchQueryBuilder
): Promise<ISearchResult<IUser>> {
  return genericSearch<IUser>(search, "user");
}

// new method added
export function searchCommunityUsers(
  search: IUserSearchOptions | SearchQueryBuilder
): Promise<ISearchResult<IUser>> {
  return genericSearch<IUser>(search,"communityUser");
}
```

OR

2. Refactor the _existing_ `searchUsers` method to accept an additional argument, with which clients can provide context into _what_ API endpoint should be called.

```ts
// option 2

export function searchUsers(
  search: IUserSearchOptions | SearchQueryBuilder,
  searchType: "user" | "communityUser" = "user"
): Promise<ISearchResult<IUser>> {
  return genericSearch<IUser>(search, searchType);
}

// OR

export function searchUsers(
  search: IUserSearchOptions | SearchQueryBuilder,
  options?: { searchType: 'user' | 'communityUser' },
): Promise<ISearchResult<IUser>> {
  return genericSearch<IUser>(search, options?.searchType ?? 'user');
}
```

### Alternatives considered

See `Describe the proposed solution` section for both options considered.

### Additional Information

Although seemingly not a large effort, I wanted to open this issue to solicit feedback from other frequent contributors to this project before submitting a PR. @dbouwman @gavinr @patrickarlt I'm curious your thoughts? Feel free to loop other contributors into this conversation.

**Comment by rweber-esri** [comment]
I chatted with @dbouwman earlier today per Hub's needs and my proposed approaches. We were in agreement that option 1 above felt more appropriate. I've submitted a pair of PRs to add a `searchCommunityUsers` method to [V3](https://github.com/Esri/arcgis-rest-js/pull/1120) & [V4](https://github.com/Esri/arcgis-rest-js/pull/1121)

**Comment by rweber-esri** [comment]
[V3 PR-1120](https://github.com/Esri/arcgis-rest-js/pull/1120) released as `v4.4.0`
[V4 PR-1121](https://github.com/Esri/arcgis-rest-js/pull/1121) released as `v3.7.0`
